server/re/src/reIn2p3SysRule.cpp

```diff
@@ -820,11 +820,15 @@ int msiFlushMonStat( msParam_t *inpParam1, msParam_t *inpParam2, ruleExecInfo_t
         return rei->status;
     }
 
-    if ( atoi( timespan ) > 0 ) {
-        elapseTime = atoi( timespan ) * 3600;
+    char *endc{};
+    if (int scanned_number = strtol( timespan, &endc, 10 ); nullptr != endc && *endc == 's') {
```

Comment:
Understood. I think we may need to move this to 4.3 only.

This PR is introducing an enhancement which doesn't feel mandatory for 4.2.12. It may also introduce more bugs due to edge cases with the seconds suffix.

Thoughts?

Reply:
If the unit is optional and the behavior does not change for existing usage, I feel like it is okay. Do you have a specific concern in mind?

One positive effect is that this MSI is now being tested, which was not the case before.

Reply:
> Understood. I think we may need to move this to 4.3 only.

@korydraughn I'm easy... it doesn't seem that critical to me that the msi work properly on 4.2.x, as those who run it are probably already well familiar with the problem and - I'm sure - run a cron job to do the same thing.

@alanking - testing and documention will both be nice-to-haves.  Which actually brings me to this: the only reason _[this test](https://github.com/irods/irods/blob/3a9c53e84a116c8f03bbbd657bc51bc53ec3e1df/scripts/irods/test/test_all_rules.py#L649)_ is not currently failing in our container based core test-suite  is that we're not using `-r irods_rule_engine_plugin-irods_rule_language-instance` in the irule invocation.

Reply:
My only hangup is shipping this with 4.2.12 and it being easy to break. The parser needs to be very good at handling invalid input such as "100 s" or "100S".

Overall, I think the idea is fine.

@d-w-moore Think through the ways someone can hold it wrong. We can discuss it tomorrow morning.

Reply:
The ways of holding it wrong are, as far as I can think of and haven't allowed for yet: 
   * too small or large an integer, that is negative or greater than `numeric_limits<int>::max()`
   * use of some bizarre form of whitespace  (`"\f"`, `"\r"`, etc)
   * units other than (s)econds, ie. assuming we offer (m)inutes as well

Reply:
If you can detect invalid input and reject it, then I think we will be in a good spot to ship this in 4.2.12.

Try and make it very strict about what it will accept. That way, no one can hold it wrong.

Reply:
Probably not for 4.2.12 unless it's trivial to implement correctly.

Years seems excessive and days can be defined in terms of hours.

Reply:
yep, ok thanks.